knowledge_storm/encoder.py

```diff
@@ -1,144 +1,158 @@
-import requests
 import os
-from typing import List, Tuple, Union, Optional, Dict, Literal
 import numpy as np
 
 from concurrent.futures import ThreadPoolExecutor, as_completed
-
-
-class EmbeddingModel:
-    def __init__(self):
-        pass
-
-    def get_embedding(self, text: str) -> Tuple[np.ndarray, int]:
-        raise Exception("Not implemented")
-
-
-class OpenAIEmbeddingModel(EmbeddingModel):
-    def __init__(self, model: str = "text-embedding-3-small", api_key: str = None):
-        if not api_key:
-            api_key = os.getenv("OPENAI_API_KEY")
-
-        self.url = "https://api.openai.com/v1/embeddings"
-        self.headers = {
-            "Content-Type": "application/json",
-            "Authorization": f"Bearer {api_key}",
-        }
-        self.model = model
-
-    def get_embedding(self, text: str) -> Tuple[np.ndarray, int]:
-        data = {"input": text, "model": self.model}
-
-        response = requests.post(self.url, headers=self.headers, json=data)
-        if response.status_code == 200:
-            data = response.json()
-            embedding = np.array(data["data"][0]["embedding"])
-            token = data["usage"]["prompt_tokens"]
-            return embedding, token
+from typing import List, Tuple, Union, Optional, Dict, Literal
+from pathlib import Path
+
+try:
+    import warnings
+
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", category=UserWarning)
+        if "LITELLM_LOCAL_MODEL_COST_MAP" not in os.environ:
+            os.environ["LITELLM_LOCAL_MODEL_COST_MAP"] = "True"
+        import litellm
+
+        litellm.drop_params = True
+        litellm.telemetry = False
+
+    from litellm.caching.caching import Cache
+
+    disk_cache_dir = os.path.join(Path.home(), ".storm_local_cache")
+    litellm.cache = Cache(disk_cache_dir=disk_cache_dir, type="disk")
+
+except ImportError:
+
+    class LitellmPlaceholder:
+        def __getattr__(self, _):
+            raise ImportError(
+                "The LiteLLM package is not installed. Run `pip install litellm`."
+            )
+
+    litellm = LitellmPlaceholder()
+
+
+class Encoder:

```

Comment:
Could we add a comment for this class? While most people know embedding models, I am unsure whether people can get the term `Encoder` easily.